eclass/python-utils-r1.eclass

```diff
@@ -1260,17 +1259,26 @@ _python_check_occluded_packages() {
 		[[ ${fn} == *.dist-info || ${fn} == *.egg-info ]] && continue
 
 		if [[ -d ${fn} ]]; then
-			diff=$(diff -dupr -x "__pycache__" "${fn}" "${sitedir}/${fn}")
+			(
+				cd "${sitedir}" && find "${fn}" -type f |
+					grep -v /__pycache__/ | sort
+			) > "${T}"/py-installed || die
+			(
+				find "${fn}" -type f |
+					grep -v /__pycache__/ | sort
+			) > "${T}"/py-source || die
+
+			diff=$(comm -1 -3 "${T}"/py-source "${T}"/py-installed || die)
 			if [[ -n ${diff} ]]; then
 				eqawarn "The directory ${fn} occludes package installed for ${EPYTHON}."
-				echo
-				echo ">>> Diff:"
-				echo "${diff}"
-				echo "<<< End-of-diff"
-				echo
+				eqawarn "The installed package includes additional files:"
+				eqawarn
+				while IFS= read -r l; do
+					eqawarn "    ${l}"
+				done <<<"${diff}"
+				eqawarn
```

Comment:
fwiw *could* skip tmp files and the read loop with this chthonic horror:
```bash
diff=$(
	comm -1 -3 \
	<(
		find "${fn}" -type f \
			-not -path '*/__pycache__/*' -printf '    %p\n' | sort
	) \
	<(
		cd "${sitedir}" && find "${fn}" -type f \
			-not -path '*/__pycache__/*' -printf '    %p\n' | sort
	) || die
)
if [[ -n ${diff} ]]; then
	eqawarn "The directory ${fn} occludes package installed for ${EPYTHON}."
	eqawarn "The installed package includes additional files:"
	eqawarn
	eqawarn "${diff}"
	eqawarn
```
(barely tested but I think(?) it should be right, output is identical on the non-fixed adblock for me)

Adapt if prefer to keep grep or whatever.

wrt tmp files, if kept, I do feel they could be bit more namespaced than already are (tend to think eclasses tmpfiles should prefix with the full eclass name given ${T} shared with ebuilds and other eclasses, but well guess chances to use py-source are still somewhat low)

Reply:
...assuming not overlooking anything, guess could optionally toss assert after sort too

unlike grep, -path is not going to return 1 if nothing matched

Reply:
Thanks, I've merged most of it. I couldn't figure out how to pass data from variables to `comm` but for some reason I didn't think of directing `find` output straight there. I just left explicit `read` loop for eqawarn, so that it is called once per file.